### ISSUE TITLE
[Defect] ✖️ button on 1Password breaks FastAuth signin

### ISSUE THREAD

**Issue by ewiner** [near/fast-auth-signer#55]
**Describe the bug**

1Password recently launched Passkey support (https://blog.1password.com/save-sign-in-passkeys-1password/) in their Chrome extension.  In Chrome on MacOS, this hijacks Chrome's WebAuthn dialog and shows a (confusing IMO) 1Password dialog in the top right corner.  It has three options:

![image](https://github.com/near/fast-auth-signer/assets/603747/7adec2a3-e7b5-4447-b197-8c1ee9211afd)

1. **Save(*: creates a new passkey in your 1Password vault (this works)
2. **Tiny USB icon in the corner**: lets you use the native Chrome WebAuthn/Passkey dialog (works)
3. ✖️: Shows this error:
![image](https://github.com/near/fast-auth-signer/assets/603747/02ab5768-2876-4ed4-850b-164e422b3510)
 
The full error in the JS console is:
```
TypeError: Cannot read properties of null (reading 'rawId')
    at oI.ot [as validateCreateRequest] (242-b3bfcd308214d60a.js:75:4581)
    at oI.parse (242-b3bfcd308214d60a.js:75:21423)
    at oI.create (242-b3bfcd308214d60a.js:75:21187)
    at async oI.create (242-b3bfcd308214d60a.js:75:21798)
    at async si.attestationResult (242-b3bfcd308214d60a.js:75:57104)
```

Note that that's different from the error you get when you cancel out of the native Chrome WebAuthn dialog:
![image](https://github.com/near/fast-auth-signer/assets/603747/4e3cf535-6ece-45f5-8008-ffaf4872f0b4)
(FYI that error message ☝️  is cut off and the link is neither clickable nor copyable.)

**Steps to Reproduce**

1. Install and set up the 1Password Chrome extension.
2. Go to near.org and click Sign In.
3. Enter your email address and click Continue.
4. Click the ✖️  in the 1Password modal.

**Expected behavior**

You should see the same "operation either timed out or was not allowed" message.  (Can we detect the difference between a timeout and not allowed?)

**Desktop**:
* OS: MacOS Ventura
* Browser: Chrome
* Version: 116.0.5845.187/ARM
* 1Password Extension: 2.15.1

**Comment by trechriron** [comment]
Outstanding defect!! :-)

**Comment by hcho112** [comment]
I was able to replicate this. Looks like 1password is hijacking the UI upon executing the 
`navigator.credentials.create..` code in browser javascript.

I'm reviewing options and also left a support post on 1password community. Will continue when I receive some feedback. 

Meanwhile,  users can continue using fast-auth with 1password by manually disabling passkey feature:
![image](https://github.com/near/fast-auth-signer/assets/6027014/1ad28470-6e6f-4130-8de4-8c38d2033229)

**Comment by ewiner** [comment]
Are we handling a `null` return value from that function? [MDN](https://developer.mozilla.org/en-US/docs/Web/API/CredentialsContainer/create#return_value) says it's allowed to return `null`, "if a single credential cannot be successfully created".

**Comment by hcho112** [comment]
@ewiner , no we aren't handling the `null` return value at the moment. However, in this case, in order to proceed to create an account, we must have `FederatedCredential` as response. 

So yes we do need to handle the `null` response. So far there could be two cases:
1. Genuine issue with browser, unable to create passkeys
2. 1Password (or any browser extensions hijacking the creation of passkeys) and clicking `X`button
![image](https://github.com/near/fast-auth-signer/assets/6027014/8dfc1f2f-cecb-4e5d-bded-07aaf423c49d)

I guess quick fix that I can come up with is use javascript `alert` to post some message. (Or we could actually use more polished UI popup) 
To handle case 1, we should show a button or fallback of creating account with storing FAK in localStorage approach. 
For case 2, we could simply say:
`Click this button` 
![image](https://github.com/near/fast-auth-signer/assets/6027014/e697f105-dd92-4e70-b45d-ea4da50e268c)
(If clicked, it redirects to a proper create passkey UI as below:
<img width="502" alt="image" src="https://github.com/near/fast-auth-signer/assets/6027014/0eddde24-cf83-4208-9004-3e3bfb66efc3">

@diogoortega @nall-near Do you have preferred way to handle this UX?

**Comment by ewiner** [comment]
As a first cut, I think this just needs an error prompt, similar to the one you get when 1password isn't installed & you cancel out the native passkey setup prompt.

I like the idea of priming the user for the passkey approval dialog (whether it's the native one or 1password's), but I think that's a separate issue that would need more evaluation:
1. Should we prime the user only after they click the ✖️, or do it before the initial request?
1. What percentage of users hit Cancel on the native dialog?
2. What percentage of users hit ✖️  on the 1Password dialog?  Can we detect client-side if 1Password is active?

**Comment by nall-near** [comment]
- We will soon have a screen in the background where we can surface a message when the passkey creation fails:
![passkey_modal](https://github.com/near/fast-auth-signer/assets/103140483/245e9b0c-ec61-41f7-95d5-3780ece01e52)
- I reproduced the steps using [webauthn.io](https://webauthn.io/), and they have a clean implementation that opens the native passkey prompt if the user clicks the✖️ on 1Password 
@esaminu , any other considerations we might be missing?

**Comment by ewiner** [comment]
> I reproduced the steps using [webauthn.io](https://webauthn.io/), and they have a clean implementation that opens the native passkey prompt if the user clicks the✖️ on 1Password

Really?  I got this error clicking the ✖️  on webauthn.io:
![image](https://github.com/near/fast-auth-signer/assets/603747/a4ea9927-3b15-4d9a-b483-27696266074a)

**Comment by nall-near** [comment]
For me it opens the native passkey prompt:
![webauth](https://github.com/near/fast-auth-signer/assets/103140483/39704336-426b-4c6d-aee0-0d5211aef6f0)

**Comment by hcho112** [comment]
> For me it opens the native passkey prompt: ![webauth](https://user-images.githubusercontent.com/103140483/273955085-39704336-426b-4c6d-aee0-0d5211aef6f0.gif) [ ![webauth]

------------
@nall-near 
You are seeing native passkey prompt because you did click edit button before hitting x button?
![image](https://github.com/near/fast-auth-signer/assets/6027014/7dbf8dfa-4a63-4e04-8b69-2db87e5fe719)
![image](https://github.com/near/fast-auth-signer/assets/6027014/27500d48-f632-4c65-be98-a6ee21558721)
<img width="453" alt="image" src="https://github.com/near/fast-auth-signer/assets/6027014/447a016c-4a58-4009-9363-eba640d44725">

It also works the same way with current FastAuth where if user clicks this pencil/edit (or native passkey button? don't know how to call this..) native selection shows up. 

Meanwhile, should I leave this issue and revisit later or without spending too much time, quickly work on handling `null` where:
if user faces 1password prompt and click `x` button -> 
show up a browser native alert with message something like `Passkey process was cancelled. `.  Something simple and quick?

**Comment by nall-near** [comment]
@hcho112 , when I press the X on 1pass, It reverts to the native passkey key flow. This is the preferred and expected behavior and seems to be in an upcoming stable release for 1Pass.

If the passkey process is cancelled or times out entirely, we should update the error message:
> Passkey process was cancelled, retry to continue account setup.

**Comment by hcho112** [comment]
This issue has been resolved as following:
- Currently on 1Password, clicking `x` button will redirect back to browser native passkey UI
- Null handling work has been implemented https://github.com/near/near-api-js/pull/1191
